### ISSUE TITLE
Can't install the library on react 18.1 because of npm not able to resolve the dependency tree

### ISSUE THREAD

**Issue by nicolinho22** [callstack/react-native-testing-library#1409]
## Describe the bug

When running npm install in the native expo project, a "unable to resolve dependency tree" error appears, blocking me from installing the dependencies. This also happens when I try to install reassure dependencies in my own project. I tried older versions as well to no avail.

## Expected behavior

Install the library successfully

## Steps to Reproduce

1. Add react 18.1
2. Try to install the library

## Screenshots

<img width="667" alt="Screenshot 2023-05-10 at 12 03 38" src="https://github.com/callstack/react-native-testing-library/assets/40872253/a323c8bd-8fc8-4f4c-ab48-60e24b1112ce">

## Versions

  npmPackages:
    react: 18.1.0 => 18.1.0
    react-native: 0.70.5 => 0.70.5

**Comment by thymikee** [comment]
Yet another issues coming from npm automatically resolving peer dependencies. I guess we should move to `peerDependenciesMeta`
https://github.com/callstack/react-native-testing-library/blob/dab444d6ddb0ad24e2d2a1fd8688761e37e008de/package.json#L66-L71

**Comment by mdjastrzebski** [comment]
@nicolinho22 It seems that you have a version mismatch between `react` (18.1.0) and `react-test-renderer` (18.2.0). These packages should be have the same version. Since you are using `react-native` version 0.70.5, so that should be 18.1.0 for both `react` and `react-test-renderer`.

**Comment by nicolinho22** [comment]
@mdjastrzebski Nope, I have react-test-renderer installed at 18.1.0 in the project correctly. I wouldn't be able to install react-test-renderer at 18.2.0 anyway because npm would complain about version mismatch.

**Comment by nicolinho22** [comment]
@mdjastrzebski Ah, this does work indeed :) Didn't expect it has to work this way. Thanks!
